mit_frontend/src/routers/Signup.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import {Card, CardTitle} from 'material-ui/Card';
+import TextField from 'material-ui/TextField';
+import RaisedButton from 'material-ui/RaisedButton';
+import '../App.css';
+import styles from './styles';
+
+class Signup extends Component {
```

Comment:
I'm not an expert of frontend, so please help me to understand how your code validate username and password, e.g. username already exists, password have invalid characters etc.

Where in the code will call Joseph's creating user profile and inserting into DB?

Reply:
This will be a software architecture design discussion. My proposal is for both sign up and sing in:

- Frontend handles the invalid characters. Because this does not require access to the DB, Frontend can reduce Backend's workload.
- Backend validates the username/password, because they require access to the DB.